pool/transaction.go

```diff
@@ -73,6 +73,18 @@ func (zkc *ZkCounters) SumUpZkCounters(txZkCounters ZkCounters) {
 	zkc.UsedSteps += txZkCounters.UsedSteps
 }
 
+// SubZkCounters subtract zk counters with passed zk counters (not safe)
+func (zkc *ZkCounters) SubZkCounters(txZkCounters ZkCounters) {
```

Comment:
I'm not very happy about this (because for how it used to be, not your changes). Things that IMO could imporve current situation:

- Move the struct to `state` package
- Ideally reuse the counters from the proto interface so we don't need to reparse (maybe by wrapping)
- `ZkCounters` => `ZKCounters` :pray: 
- Include the underflow logic already in the `SubZkCounters` method
- `SubZkCounters` => `Sub`

Reply:
The only actual request here is the renamings of the functions, otherwise it's just suggestions up to you to implement if you consider them legit